src/server/journal/types.h

```diff
@@ -44,6 +48,35 @@ struct Entry {
   uint64_t expire_ms = 0;  // 0 means no expiry.
 };
 
+// This struct represents a single journal entry.
+// Those are either control instructions or commands.
+struct EntryNew {  // Called this "New" because I can't delete the old neither replace it partially
+  // Payload represents a non-owning view into a command executed on the shard.
+  using Payload =
+      std::variant<std::monostate,                        // No payload.
+                   CmdArgList,                            // Parts of a full command.
```

Comment:
can you give example when CmdArgList is set , and why only parts of the full command enter the payload
and when the std::pair<std::string_view, ArgSlice> is set
Why do we need this 2 options?

Reply:
CmdArgList is for single shard commands, the pair for a combination of command name shard args.

Please look at how it's used in my original PR

Reply:
Ok I believe you can simplify this by using only std::pair<std::string_view, ArgSlice> and for single shard commands the ShardArgsInShard will return all args of the command, the command name is stored in Transaction class cid_ var I think, this way you dont need full_args_ .

Reply:
DF has a weird split, where multi shard commands use shard args for all values, and single shard commands use it only for determining the shard to run on.

For single shard commands it contains only the key, but not the full data. Changing this internal behavior is possible but requires revisiting transaction code and all the commands

Reply:
Ok. If this will simplify the code you could still use full_args_ and only std::pair<std::string_view, ArgSlice>  init with full_args_.front(), full_args.Span(1) . If you think this makes the flow simpler...